spot-contracts/contracts/_utils/TrancheManager.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import { IERC20Upgradeable, IBondController, ITranche } from "../_interfaces/IPerpetualTranche.sol";
+
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { MathUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/math/MathUpgradeable.sol";
+import { BondTranches, BondTranchesHelpers } from "./BondTranchesHelpers.sol";
+import { TrancheHelpers } from "./TrancheHelpers.sol";
+import { BondHelpers } from "./BondHelpers.sol";
+
+/**
+ *  @title TrancheManager
+ *
+ *  @notice Linked external library with helper functions for tranche management.
+ *
+ *  @dev Proxies which use external libraries are by default NOT upgrade safe.
+ *       We guarantee that this linked external library will never trigger selfdestruct,
+ *       and thus one is.
+ *
+ */
+library TrancheManager {
+    // data handling
+    using BondHelpers for IBondController;
+    using TrancheHelpers for ITranche;
+    using BondTranchesHelpers for BondTranches;
+
+    // ERC20 operations
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    // math
+    using MathUpgradeable for uint256;
+
+    //--------------------------------------------------------------------------
+    // Helper methods
+
+    /// @notice Low level method which tranches underlying tokens into the provided bond contract.
+    function execTranche(IBondController bond, IERC20Upgradeable underlying_, uint256 underlyingAmt) public {
```

Comment:
This function is maybe better placed in BondHelpers.sol? Seems like a library function that would be called with the `using` pattern. It already takes the bond as the first argument.

Reply:
Structurally that makes sense, but moving it there wont help trim the bytecode